tests/basic/sharedv4_svc_test.go

```diff
@@ -345,6 +345,136 @@ var _ = Describe("{Shared4SvcFailoverIO}", func() {
 	})
 })
 
+// Bring PX down on the node where the volume is attached. Verify that pods on client nodes unmount the volume
+// and teardown successfully
+var _ = Describe("{Sharedv4ClientTeardownWhenServerOffline}", func() {
+	var testrailID = 54780
+	// testrailID corresponds to: https://portworx.testrail.net/index.php?/cases/view/54780
+	var runID, numPods int
+	var testSv4Contexts, contexts []*scheduler.Context
+
+	JustBeforeEach(func() {
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
+	})
+
+	It("has to schedule apps, stop volume driver on node where volume is attached, teardown the application", func() {
+		contexts = make([]*scheduler.Context, 0)
+
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("clientteardown-%d", i))...)
+		}
+		testSv4Contexts = getTestSv4Contexts(contexts)
+
+		Step("scale the sharedv4 apps so that one pod runs on each worker node", func() {
+			if len(testSv4Contexts) == 0 {
+				Skip("No sharedv4 apps were found")
+			}
+			numPods = len(node.GetWorkerNodes())
+			for _, ctx := range testSv4Contexts {
+				Step(fmt.Sprintf("scale up app %s to %d", ctx.App.Key, numPods), func() {
+					applicationScaleUpMap, err := Inst().S.GetScaleFactorMap(ctx)
+					Expect(err).NotTo(HaveOccurred())
+					for name := range applicationScaleUpMap {
+						applicationScaleUpMap[name] = int32(numPods)
+					}
+					err = Inst().S.ScaleApplication(ctx, applicationScaleUpMap)
+					Expect(err).NotTo(HaveOccurred())
+				})
+			}
+
+			ValidateApplications(testSv4Contexts)
+		})
+
+		var attachedNode *node.Node
+		Step("stop the volume driver on attached node and verify application teardown succeeds", func() {
+			for _, ctx := range testSv4Contexts {
+				vols, err := Inst().S.GetVolumes(ctx)
+				Expect(err).NotTo(HaveOccurred())
+
+				attachedNode, err = Inst().V.GetNodeForVolume(
+					vols[0],
+					defaultCommandTimeout,
+					defaultCommandRetry,
+				)
+				Expect(err).NotTo(HaveOccurred())
+
+				Step(
+					fmt.Sprintf("stopping volume driver on node %s", attachedNode.Name),
+					func() {
+						StopVolDriverAndWait([]node.Node{*attachedNode})
+					},
+				)
+
+				Step(fmt.Sprintf("scale down app %s to 0", ctx.App.Key), func() {
```

Comment:
Could the failover have happened at this point?  This test is good, but wondering if there should also be a test that scales down app while there was no failover yet (perhaps by switching to "normal" failover strategy).

Reply:
Yes I was also thinking of that. While testing I did see a few occurrences where failover happened even before the scale down. Doing a failover and then scale down would probably get executed as a part of other tests too. So let me change the storage class / update the volume to use normal strategy.

Reply:
Added steps to change the failover strategy to normal